internal/weeder/weeder.go

```diff
@@ -20,7 +21,7 @@ const crashLoopBackOff = "CrashLoopBackOff"
 // are in CrashLoopBackOff.
 type Weeder struct {
 	namespace          string
-	endpoints          *v1.Endpoints
+	endpoints          *discoveryv1.EndpointSlice
```

Comment:
can we rename this to `endpointSlice` since `Endpoints` is a field inside `EndpointSlice` and can get confusing.